net/respondd-wifi/src/respondd.c

```diff
@@ -0,0 +1,104 @@
+#include <string.h>
+#include <stdio.h>
+#include <json-c/json.h>
+#include <respondd.h>
+
+#include <linux/nl80211.h>
+
+#include "clients.h"
+#include "neighbours.h"
+#include "ifaces.h"
+
+static struct json_object *respondd_provider_statistics(void) {
+	struct json_object *result, *clients;
+	struct iface_list *ifaces;
+	int wifi24 = 0, wifi5 = 0, clients_count;
+
+	result = json_object_new_object();
+	if (!result)
+		return NULL;
+
+	clients = json_object_new_object();
+	if (!clients) {
+		json_object_put(wireless);
+		json_object_put(result);
+		return NULL;
+	}
+
+	ifaces = get_ifaces();
+	while (ifaces != NULL) {
+		if(ifaces->type != NL80211_IFTYPE_AP)
+			goto next_statistics;
+		
+		clients_count = 0;
+		get_client_counts(&clients_count, ifaces->ifx);
+		if (ifaces->frequency < 5000)
+			wifi24 += clients_count;
+		if (ifaces->frequency > 5000)
+			wifi5 += clients_count;
+next_statistics: ;
+		void *freeptr = ifaces;
+		ifaces = ifaces->next;
+		free(freeptr);
+	}
+
+	//TODO maybe skip: if (wifi24 >  0 || wifi5 > 0) {
+	json_object_object_add(clients, "wifi24", json_object_new_int(wifi24));
+	json_object_object_add(clients, "wifi5", json_object_new_int(wifi5));
+	json_object_object_add(result, "clients", clients);
```

Comment:
Hmm, I'm not sure if it really makes sense to count the clients in a generic respondd module after all. Since freifunk-gluon/gluon@b850fff7e4f4b938c512d81de31bfae872fa42e4, "wifi" is computed from "wifi24" and "wifi5" as well - which would need to be moved into this module as well; but the computation of "total" depends on "wifi" now.

A different approach would be to move this query into a reusable library included by both gluon-mesh-... respondd providers...